contracts/proxies/ProxyFactory.sol

```diff
@@ -23,4 +23,37 @@ contract ProxyFactory {
             }
         emit ProxyCreation(proxy);
     }
+
+    /// @dev Allows to retrive the runtime code of a deployed Proxy. This can be used to check that the expected Proxy was deployed.
+    function proxyRuntimeCode() public pure returns (bytes memory) {
+        return type(Proxy).runtimeCode;
+    }
+
+    /// @dev Allows to retrive the creation code used for the Proxy deployment. With this it is easily possible to calculate predicted address.
+    function proxyCreationCode() public pure returns (bytes memory) {
+        return type(Proxy).creationCode;
+    }
+
+    /// @dev Allows to create new proxy contact and execute a message call to the new proxy within one transaction.
+    /// @param _mastercopy Address of master copy.
+    /// @param initializer Payload for message call sent to new proxy contract.
+    /// @param nonce Nonce that will be used to generate the salt to calculate the address of the new prox contract.
+    function createProxyWithNonce(address _mastercopy, bytes memory initializer, uint256 nonce)
```

Comment:
I still don't like `nonce` as name for the parameter to calculate the salt... Maybe it's a little confusing because of the safe `nonce` and ethereum tx `nonce`

Reply:
any idea for a better name? ... I didn't want to use `salt` since that will be derived. I was thinking about something with `random`, but in the end I though everybody knows what nonces are used for.

Reply:
Is the nonce required to create 2 Safes with identical payload?

Reply:
To make the parameter more descriptive we could rename it to`saltNonce`.